### ISSUE TITLE
We need an option to stop running when the first benchmark fails.

### ISSUE THREAD

**Issue by jorive** [dotnet/BenchmarkDotNet#939]
When running benchmarks in the lab, it is ideal to fail fast and avoid using resources unnecessarily. When a benchmark fails we would like to acknowledge the developers as soon as possible.
Currently, the harness keeps executing all benchmarks to completion. Is there a way to tell BDN to fail fast?

**Comment by adamsitnik** [comment]
I think that it would be nice to add such feature. 

It could be added to `IConfig` as `StopOnFirstError` with default value = false and exposed via console line arguments to override.

@wojtpl2 maybe you would like to give it a try?

**Comment by wojtpl2** [comment]
Ok next week

**Comment by jorive** [comment]
@adamsitnik What is the reasoning for `default value = false`?

**Comment by adamsitnik** [comment]
@jorive to don't change the current default behavior (breaking change). I like the current one: last night I started the benchmark run for all the benchmarks we have and today in the morning I had the results for few thousands of them. 8 failed had no results but I had the errors in the logs. If it would stop on the first error I would have to fix the problem and rerun all the benchmarks again. With current behavior, I just fixed and re-run 8 of them.
